querybook/server/tasks/log_query_per_table.py

```diff
@@ -80,46 +80,35 @@ def create_lineage_from_query(
 def sync_table_to_metastore(
     table_per_statement, statement_types, metastore_id, session=None
 ):
+    """Sync tables with metastore. Tables are parsed from executed queries.
+
+    It syncs below two kinds of tables:
+        - tables from CREATE, ALTER, DROP statements.
+        - tables from other statements, but they dont exsit in Querybook database.
+    """
     metastore_loader = get_metastore_loader(metastore_id, session=session)
     assert metastore_loader is not None
 
-    tables_to_add = set()
-    tables_to_remove = set()
+    tables_to_sync = set()
     for tables, statement_type in zip(table_per_statement, statement_types):
-        if statement_type == "DROP":
+        if statement_type in ("CREATE", "ALTER", "DROP"):
             for table in tables:
-                tables_to_add.discard(table)
-                tables_to_remove.add(table)
-        elif statement_type is not None:  # Any other DML/DDL
+                tables_to_sync.add(table)
+        elif statement_type is not None:
```

Comment:
keep the comments since they are still applicable
```
# Otherwise for things like insert/select we only update
# if it doesn't exist in the metastore